zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterServer.java

```diff
@@ -340,7 +344,21 @@ private void putClusterMeta() {
     meta.put(ClusterMeta.LATEST_HEARTBEAT, LocalDateTime.now());
     meta.put(ClusterMeta.STATUS, ClusterMeta.ONLINE_STATUS);
 
-    clusterManagerClient.putClusterMeta(ClusterMetaType.INTP_PROCESS_META, interpreterGroupId, meta);
+    clusterManagerClient.putClusterMeta(INTP_PROCESS_META, interpreterGroupId, meta);
+  }
+
+  private void deleteClusterMeta() {
+    if (!zconf.isClusterMode()){
+      return;
+    }
+
+    try {
+      // delete interpreter cluster meta
+      clusterManagerClient.deleteClusterMeta(INTP_PROCESS_META, interpreterGroupId);
+      Thread.sleep(300);
```

Comment:
why a sleep here?

Reply:
When `clusterManagerClient` operation metadata operations, it is asynchronous, so it can be more reliable.